src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol.cs

```diff
@@ -821,6 +821,22 @@ internal override bool HasSpecialName
             }
         }
 
+        internal override bool HasInternalImplementationOnlyAttribute
+        {
+            get
+            {
+                foreach (var att in this.GetAttributes())
+                {
+                    if (att.IsTargetAttribute("System.Runtime.CompilerServices", "InternalImplementationOnlyAttribute"))
```

Comment:
This is how all well-known attributes are matched even if the spec doesn't talk about that explicitly (ExtensionAttribute, etc.) I don't see any good reason to make an exception for this attribute.